projects/gateway2/setup/ggv2setup_test.go

```diff
@@ -312,22 +348,110 @@ func testScenario(t *testing.T, ctx context.Context, client istiokube.CLIClient,
 	fmt.Println("test done")
 }
 
-func getXdsDump(t *testing.T, ctx context.Context, xdsPort int, gwname string) xdsDump {
-	conn, err := grpc.NewClient(fmt.Sprintf("localhost:%d", xdsPort), grpc.WithTransportCredentials(insecure.NewCredentials()))
+type xdsDumper struct {
```

Comment:
This already existed, so I'm ok to leave it here. It seems like a pretty useful utility that others could benefit from it it lived in a more standardized place. We've been building some utilities out for envoy here: https://github.com/solo-io/gloo/tree/main/pkg/utils/envoyutils. An alternative would be in the xds package: https://github.com/solo-io/gloo/tree/main/projects/gloo/pkg/xds

Reply:
This code is very specialized for this use case - this is getting xds snapshot by imitating envoy pod (as this triggers the final step of translation, so we can assert on the impact of DesetinationRules). 
I would assume that other utilities will interact with a real server with envoy connected to it, and they can probably check the `/snapshots/xds` on the server instead.